### PR TITLE
Conversions: Allow "celcius" typo, updated tests

### PR DESCRIPTION
**Pull request by moollaza**
## Description of new Instant Answer, or changes
Many queries are made with the word celsius misspelled as celcius. Seeing as we already account for the "farenheit" misspelling, I've added in "celcius" to account for these missed opportunities.

## People to notify
@pjhampton 

## Testing & Review
To be completed by Community Leader (or DDG Staff) when reviewing Pull Request

**Pull Request**
- [ ] Title follows correct format (Specifies Instant Answer + Purpose)
- [ ] Description contains a valid Instant Answer Page Link (e.g. https://duck.co/ia/view/my_ia)

**Code**
- [ ] Adheres to the [DuckDuckGo Style Guide](https://docs.duckduckhack.com/resources/code-style-guide.html)
- [ ] Behaviour is appropriately tested. If improvement, tests are adequately extended.
- [ ] There is no unnecessary files in place (such as editor config files)
- [ ] There is no API keys / secrets present
- [ ] Tests are passing (run `$ duckpan test <goodie_id>`)
    - Tester should report any failures

**Ready to merge?**

- [ ] Has this IA been deployed to and tested on [beta.duckduckgo.com](https://beta.duckduckgo.com/)?
- [ ] For larger commits, has this been approved be more than one community member?
- [ ] The number of reviews is appropriate for this type of PR
- [ ] The commit message is clear, coherent and fitting

**Pull Request Review Guidelines**: https://docs.duckduckhack.com/programming-mission/pr-review.html

---

Instant Answer Page: https://duck.co/ia/view/conversions

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: t/Conversions.t [hunk 1]
```diff
@@ -226,6 +226,24 @@ ddg_goodie_test(
 			physical_quantity => 'temperature'
 		})
 	),
+	## with typos
+	'convert 5 kelvin to farenheit' => test_zci(
+		'', structured_answer => make_answer({
```
Comment (kirkins) [inline t/Conversions.t]: This goodie always returns an empty string for the plain text answer? 

Is the plain text answer for goodies used anywhere or is it just legacy at this point?

Comment (mintsoft) [inline t/Conversions.t]: They're returned by the API, on https://duckduckgo.com/html (ironically my preference to use ddg) and https://duckduckgo.com/lite. However without moving all the conversion logic back into the backend, there's no way of implementing an actual answer

File: lib/DDG/Goodie/Conversions.pm [hunk 2]
```diff
@@ -96,7 +96,7 @@ handle query => sub {
     }
 
     # hack support for "degrees" prefix on temperatures
-    $_ =~ s/ degree[s]? (centigrade|celsius|fahrenheit|rankine)/ $1/i;
+    $_ =~ s/ degree[s]? (centigrade|cel(?:s|c)ius|fah?renheit|rankine)/ $1/i;
```
Comment (mintsoft) [inline lib/DDG/Goodie/Conversions.pm]: Could do `cel[sc]ius` and personally I prefer `fa[h]?renheit` but I'm not that bothered

File: t/Conversions.t (hunk 1 above)
Comment (kirkins) [inline t/Conversions.t]: If you look [further back](https://github.com/duckduckgo/zeroclickinfo-goodies/commit/c42daedc1e27c6ded912209dc99d8fd02e9cc2eb#diff-4f3dd19e406358035d38859511d702d1) in the history it use to return string answers as well.

Comment (mintsoft) [inline t/Conversions.t]: That's from when it used to convert the units in the perl. We've moved it all to the frontend (js) since then

Comment (kirkins) [inline t/Conversions.t]: @mintsoft seems to be the trend around here. If you move something from Perl to mah JS it's considered an "improvement". When everything is powered by the front-end the tests essentially become meaningless.

Comment (mintsoft) [inline t/Conversions.t]: Well, it's the cost of interactivity. There are tests around the frontend of conversions, but they're not as broad as the backend engine was; however they'll improve over time.

**Comment by moollaza** [comment]
> If you move something from Perl to mah JS it's considered an "improvement".

@kirkins as @mintsoft pointed out, this is an unfortunate but expected cost of moving to an interactive frontend, which is truly an "improvement" for the Instant Answers. Many users expect DDG to be at least on-par, with Google and other search engines, and the lack of interactivity has been a constant complaint.

It was a tough call to make, but the reality is that very few people use the API, /lite, or /html so making the Conversion IAs interactive has had an overall very positive impact with the majority of our users.

I hope that in the future we'll be able to get these IAs working through the API, but for now it's not something we're focused on.

**Pull request merged**